### ISSUE TITLE
about heightmaps, must this be considered a bug?

### ISSUE THREAD

**Issue by earthling45** [TheGameCreators/GameGuruRepo#3342]
Hi, after importing a heightmap there is a sharp height difference on the edge of the editable area.
No matter how i adjust the sliders, there is no way for me to eliminate this sharp difference and although it is not really a problem because it is at the edge i do wonder if this must be considered a bug.
![level](https://user-images.githubusercontent.com/35744254/175296547-295719ca-6e32-4159-a44f-62355f75faf7.png)

**Comment by earthling45** [comment]
Hi @bbacle 

I've used Qgis to create this map and painted afterwards in paint.net. the heightdata is taken from ahn-Nederland and extracted from Qgis as a DTM (heightdata without buildings and trees) and the holes (where those removed buildings were for example) filled with a matching grey scale.
The heightmap is a PNG 16bit greyscale and in my opinion good enough quality.
If you import a rawfile you should try Little endian when importing if you get those spikes, by default Big endian is ticked.
My map is also not really smooth, but while working on the map i can use the blendtool to smooth it, though that is for a later date because i'm waiting for the possibility that custom textures can be imported. :)
![level2](https://user-images.githubusercontent.com/35744254/175382276-cbd4541c-998a-41ba-94fa-b264a39f28f0.png)

**Comment by earthling45** [comment]
For those who try out Qgis, do also look here. https://forums.bohemia.net/forums/topic/224613-qgis-game-terrain-tools/

**Comment by bbacle** [comment]
Thanks...looks interesting...will give it a try.

From: Bob Bacle
***@***.******@***.***>

________________________________

**Comment by LeeBamberTGC** [comment]
@earthling45 Yes I think we can call this a bug, as you would want a smooth transition into the procedural geometry to give a realistic look.

**Comment by LeeBamberTGC** [comment]
@earthling45 This relates to another sort-of request to have smoothing options around the edges of the terrain, and I think this fix and that feature could be the same code. Just knocking this down to LOW so I can tackle higher priority bugs, but this wil get done around the same time as that functional smoothing code.

**Comment by LeeBamberTGC** [comment]
@earthling45 Was playing with the latest heightmap importer and found a nice slider which resolves the edge issue of imported heightmaps. Here is the value set at zero and shows the edge issue:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/4b8b3a7d-1044-4142-95ad-d42b957cf6a5)

When you increase the value however, you can blend the heightmap data into the procedural terrain nicely:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/64bdca94-5cf6-465e-ae66-bd3bebb952c0)

I have tagged this as an enhancement in case we need finer grain controls in this area for specific heightmap uses.

**Comment by earthling45** [comment]
> @earthling45 Was playing with the latest heightmap importer and found a nice slider which resolves the edge issue of imported heightmaps. Here is the value set at zero and shows the edge issue: ![image](https://user-images.githubusercontent.com/35840583/280058911-4b8b3a7d-1044-4142-95ad-d42b957cf6a5.png)
> 
> When you increase the value however, you can blend the heightmap data into the procedural terrain nicely: ![image](https://user-images.githubusercontent.com/35840583/280059065-64bdca94-5cf6-465e-ae66-bd3bebb952c0.png)
> 
> I have tagged this as an enhancement in case we need finer grain controls in this area for specific heightmap uses.

Hi Lee, i remember that slider, but it did not produce proper results as you can see in my previous reaction whereas the slider was used and the result shown in those images which i posted.

I think i have to redo the heightmap and have the base color properly set, it is after all a heightmap with black being the lowest and white being the highest. 
So if we divide 255 we get 127,5 and this would be 0 or base level/color.

Though i of course would like to know your opinion on this. :)
